includes/class-wcpdf-admin.php

```diff
@@ -790,150 +793,47 @@ public function get_current_values_for_document( $document, $data ) {
-				<div class="read-only">
-					<span><strong><?= wp_kses_post( $data['notes']['label'] ); ?></strong></span>
-					<?php if ( $this->user_can_manage_document( $document->get_type() ) ) : ?>
-						<span class="wpo-wcpdf-edit-document-notes dashicons dashicons-edit" data-edit="notes"></span>
-					<?php endif; ?>
-					<p><?= ( $data['notes']['value'] == strip_tags( $data['notes']['value'] ) ) ? wp_kses_post( nl2br( $data['notes']['value'] ) ) : wp_kses_post( $data['notes']['value'] ); ?></p>
-				</div>
-				<!-- Editable -->
-				<div class="editable-notes">
-					<p class="form-field form-field-wide">
-						<label for="<?= esc_attr( $data['notes']['name'] ); ?>"><?= wp_kses_post( $data['notes']['label'] ); ?></label>
-						<p><textarea name="<?= esc_attr( $data['notes']['name'] ); ?>" class="<?= esc_attr( $data['notes']['name'] ); ?>" cols="60" rows="5" disabled="disabled"><?= wp_kses_post( $data['notes']['value'] ); ?></textarea></p>
-					</p>
-				</div>
+		// Go for default data.
+		if ( empty( $data ) ) {
+			$data = array(
+				'number' => array( 'label' => __( 'Number', 'woocommerce-pdf-invoices-packing-slips' ) . ':' ),
+				'date'   => array( 'label' => __( 'Date', 'woocommerce-pdf-invoices-packing-slips' ) . ':' ),
+			);
 
-				<?php do_action( 'wpo_wcpdf_meta_box_after_document_notes', $document, $document->order ); ?>
+			if ( 'invoice' === $document->get_type() ) {
+				$data = array_merge( $data, array(
+					'display_date'     => array( 'label' => __( 'Display date', 'woocommerce-pdf-invoices-packing-slips' ) . ':' ),
+					'creation_trigger' => array( 'label' => __( 'Created via', 'woocommerce-pdf-invoices-packing-slips' ) . ':' ),
+					'notes'            => array( 'label' => __( 'Notes (printed in the invoice)', 'woocommerce-pdf-invoices-packing-slips' ) . ':' ),
+				) );
+			}
+		}
 
-				<?php endif; ?>
-				<!-- / Document Notes -->
+		$independent_documents = apply_filters( 'wpo_wcpdf_document_data_meta_box_independent_documents', array( 'invoice', 'packing-slip' ) );
+		$data                  = apply_filters( 'wpo_wcpdf_document_data_meta_box_document_data_fields', $data, $document );
+		$data                  = $this->get_current_values_for_document( $document, $data );
+		$in_process            = as_has_scheduled_action( 'wpo_wcpdf_generate_document_on_order_status', array(
+			'document_type' => $document->get_type(),
+			'order_id'      => $document->order->get_id()
+		) );
+
+		// Prevent displaying setup for documents that are not independent, and don't exist, or are not in process,
+		// like documents that require an invoice first and don't exist already or are not in process.
+		if ( ! $in_process && ! $document->exists() && ! in_array( $document->get_type(), $independent_documents, true ) ) {
+			return;
+		}
 
-			</section>
+		// Allow preventing document output.
+		if ( apply_filters( 'wpo_wcpdf_document_data_meta_box_allow_document_output', false, $data, $in_process, $document ) ) {
```

Comment:
I find the naming convention of the filter hook `wpo_wcpdf_document_data_meta_box_allow_document_output` a bit misleading. Given its name, one might expect that setting the filter to `true` would enable the document output process. However, in practice, if the filter returns `true`, the function exits early (`return;`), effectively preventing any document output. This behavior seems counterintuitive based on the filter's name, suggesting that allowing document output actually prevents it.

Reply:
Both are good to me, maybe the first one, but I don't have a strong opinion.